root/usr/share/fdi/facts/openlldp.rb

```diff
@@ -0,0 +1,99 @@
+# Fact:
+#   lldp_neighbor_chassisid_<interface>
+#   lldp_neighbor_mngaddr_ipv4_<interface>
+#   lldp_neighbor_mngaddr_ipv6_<interface>
+#   lldp_neighbor_mtu_<interface>
+#   lldp_neighbor_portid_<interface>
+#   lldp_neighbor_sysname_<interface>
+#   lldp_neighbor_pvid_<interface>
+#
+# Purpose:
+#   Return information about the host's LLDP neighbors.
+#
+# Resolution:
+#   On hosts with the lldptool binary, send queries to the lldpad for each of
+#   the host's Ethernet interfaces and parse the output.
+#
+# Caveats:
+#   Assumes that the connected Ethernet switch is sending LLDPDUs, Open-LLDP
+#   (lldpad) is running, and lldpad is configured to receive LLDPDUs on each
+#   Ethernet interface.
+#
+# Authors:
+#   Mike Arnold <mike@razorsedge.org>
+#
+# Copyright:
+#   Copyright (C) 2012 Mike Arnold, unless otherwise noted.
+#
+
+require 'facter/util/macaddress'
+
+# http://www.ruby-forum.com/topic/3418285#1040695
+module Enumerable
+  def grep_v(cond)
+    select {|x| not cond === x}
+  end
+end
```

Comment:
Would you mind *not* extending `Enumerable` here and creating a helper method? This will be unlikely shared and Facter is fragile codebase already.